SectionRepository testai, JSON data, Json failu handlinimas(!)

pkondratas opened the pull request:
Prideti testai, dabar data nebera skirstoma i folderius, o i atskirus jsonus(Section.json, Note.json veliau ir t.t) 

Praktiskai pakeista logika CreateSection metode(SectionRepository)

Pagrindinis pakeitimas yra sukurta DataHandler klase, kuri isparsina norima duomenu faila ir suparsina atgal su metodais GetAllItems<>(), PostAllItems. Klase yra generic todel visiem tipam tinka, tik reikia nurodyti enumo verte kuria norima pasiekti ar patalpinti(Sections, Notes ir t.t.)

Taip pat pridejau ir dependency injection, kad lengviau mockint butu. Su sita klase nereikes visiem rasyt direktorijos stringo, sekciju, notu ir visu kitu repositorijos bus "atsakingos tik uz viena", o parsinsim ir postinsim su DataHandler klase. 

Enumo reiksmiu reikia nekeist, bet cia kaip ir standartas. Taip pat, visi jsonai turi buti daugiskaitos modelio vardu(Sections.json, Notes.json ir t.t)

Paziurekit ar viskas korektiskai ir gerai. Manau tikrai patogus budas.

P.S. Ten buvoau nesamoniu padares Collibri.csproj faile su <ItemGroups> kuriu nereikia. Kai sukuri tuscia folderi juos, padaro, nors kai jie turi kazka ju net nereikia, tai istryniau.

pkondratas committed:
SectionRepository testai, JSON data, Json failu handlinimas

Prideti testai, dabar data nebera skirstoma i folderius, o i atskirus jsonus(Section.json, Note.json veliau ir t.t) Sukurta DataHandler klase, kuri isparsina norima duomenu faila ir suparsina atgal su metodais GetAllItems<>(), PostAllItems. Klase yra generic todel visiem tipam tinka, tik reikia nurodyti enumo verte kuria norima pasiekti ar patalpinti(Section, Note ir t.t.)

pkondratas committed:
Pritaikytas encapsulation modelio klasei

pkondratas committed:
Pataisytas testo netikslumas, sekcijos fieldai(siam atvejui nereikia private))

dovix58 reviewed (approved):
Viskas atrodo tvarkoje, tik vat tas pastebejimas del if/else

dovix58 commented on Collibri/Models/Sections/SectionRepository.cs:
```diff
@@ -0,0 +1,31 @@
+            List<Section> sectionList = _dataHandler.GetAllItems<Section>(ModelType.Sections);
+            
+            foreach (var sections in sectionList)
+                if (sections.RoomId.Equals(section.RoomId) && sections.SectionName.Equals(section.SectionName))
+                    return null;
+            
```
Toks vat pastebejimas. Žinau, kad vienam tam if/else sakiniui nereikia {}, bet kiek skaiciau ir is kitu girdejau, kad if/else geriau visad naudoti {}.

pkondratas replied:
okey pakeisiu

pkondratas committed:
Pridetos {} pries foreach ir if

jogis0 commented on Collibri.sln.DotSettings.user:
```diff
@@ -1,4 +1,4 @@
 ﻿<wpf:ResourceDictionary xml:space="preserve" xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml" xmlns:s="clr-namespace:System;assembly=mscorlib" xmlns:ss="urn:shemas-jetbrains-com:settings-storage-xaml" xmlns:wpf="http://schemas.microsoft.com/winfx/2006/xaml/presentation">
 	<s:String x:Key="/Default/Environment/UnitTesting/UnitTestSessionStore/Sessions/=3d93ca93_002D2b63_002D4844_002Da85a_002D2d1276577364/@EntryIndexedValue">&lt;SessionState ContinuousTestingMode="0" IsActive="True" Name="All tests from UnitTest1.cs" xmlns="urn:schemas-jetbrains-com:jetbrains-ut-session"&gt;&#xD;
-  &lt;ProjectFile&gt;6D18782F-CC2C-4E91-BAEF-1B00299531D7/f:UnitTest1.cs&lt;/ProjectFile&gt;&#xD;
+  &lt;Project Location="C:\Users\Paul James\Desktop\3 semestras\PS inzinerija\Collibri\Collibri.Tests" Presentation="&amp;lt;tests&amp;gt;\&amp;lt;Collibri.Tests&amp;gt;" /&gt;&#xD;
```
Cia kaip ir tavo local direktorija yra, tai gal reiktu pakeist kazkaip, kad universaliai butu. Arba sito failo necommitint, tik nezinau tada, ar veiks testai

jogis0 commented on Collibri/Models/Sections/Section.cs:
```diff
@@ -0,0 +1,18 @@
+    public class Section
+    {
+        public int SectionId { get; set; }
+        public int RoomId { get; set; }
```
O section'e reikia roomId saugoti? Speju galima gal sita kol kas palikti (nes dabar nezinau, gal ir patogu bus turet roomId), bet jeigu ateity matysim, kad nereikia, galima bus isimt

dovix58 replied:
As kaip supratau, cia anas reikalingas, kad galetum paziuret ar butent konkreciam kambary nera tokio jau vardo, kitaip nezinosi

jogis0 reviewed (changes requested):
Manau viskas yra gerai, tik vat gal ta DotSettings reikia pakeist arba is vis necommitint, nes tada speju kitiem gal gali neveikt.

dovix58 commented on Collibri/Models/Sections/SectionRepository.cs:
```diff
@@ -0,0 +1,35 @@
+            
+            section.SectionId = new Random().Next(1, int.MaxValue);
+            sectionList.Add(section);
+            
```
Dar vat pagalvojau, kad cia nera apsaugos, kad SectionID nesikratotu. Nors praktiskai mazas sansas, bet jei daug bus sectionu gali pakliut vienodi ID'ai

pkondratas committed:
Istrintas DotSettings.user, patikslintas testas

Pasirodo Collibri.sln.DotSettings.user faila sukuria  ReSharperis ir ji reikia excludint kai mes keiciam sita repositorija gite, nes visiem jis bus skirtingas kaip suprantu. https://www.jetbrains.com/help/resharper/Sharing_Configuration_Options.html#solution-personal-layer

pkondratas commented:
Pasirodo Collibri.sln.DotSettings.user faila sukuria  ReSharperis ir ji reikia excludint kai mes keiciam sita repositorija gite, nes visiem jis bus skirtingas kaip suprantu. https://www.jetbrains.com/help/resharper/Sharing_Configuration_Options.html#solution-personal-layer

jogis0 reviewed (approved):
Dabar turbut viskas ok, I approve
